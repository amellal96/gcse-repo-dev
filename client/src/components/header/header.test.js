import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
import {Provider} from 'react-redux';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import Header from './header.component';

const mockStore = configureStore([thunk]);

describe('Header component - teacher links', () => {
    let store;
    let wrapper;
    let mockAuth;

    beforeEach(() => {
        store = mockStore({
            user: {
                isAuthenticated: true,
                loading: false,
                user: {
                    accountType: 'teacher'
                }
            }
        }); 

        store.dispatch = jest.fn();

        const mockProps = {
            store,
            mockAuth: jest.fn()
        }

        wrapper = mount(
                <Provider store={store}>
                    <Header {...mockProps} />
                </Provider>
        );
    });

    it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot(); 
    });

    it('should render teacher links in nav bar', () => {
        expect(wrapper.find('.nav-item.active')).toHaveLength(5);
    });
});

describe('Header component - student links', () => {
    let store;
    let wrapper;
    let mockAuth;

    beforeEach(() => {
        store = mockStore({
            user: {
                isAuthenticated: true,
                loading: false,
                user: {
                    accountType: 'student'
                }
            }
        }); 

        store.dispatch = jest.fn();

        const mockProps = {
            store,
            mockAuth: jest.fn()
        }

        wrapper = mount(
                <Provider store={store}>
                    <Header {...mockProps} />
                </Provider>
        );
    });

    it('should render student links in nav bar', () => {
        expect(wrapper.find('.nav-item.active')).toHaveLength(4);
    });
});

describe('Header component - guest links', () => {
    let store;
    let wrapper;
    let mockAuth;

    beforeEach(() => {
        store = mockStore({
            user: {
                isAuthenticated: false,
                loading: false,
                user: {
                    
                }
            }
        }); 

        store.dispatch = jest.fn();

        const mockProps = {
            store,
            mockAuth: jest.fn()
        }

        wrapper = mount(
                <Provider store={store}>
                    <Header {...mockProps} />
                </Provider>
        );
    });

    it('should render guest links in nav bar', () => {
        expect(wrapper.find('.nav-item.active')).toHaveLength(3);
    });
});