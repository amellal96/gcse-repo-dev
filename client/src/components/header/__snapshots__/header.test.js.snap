// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component - teacher links should match the snapshot 1`] = `"<div class=\\"navbar navbar-expand-lg navbar-dark bg-primary\\"><a class=\\"navbar-brand\\" href=\\"/\\">GCSE Repo</a><div class=\\"collapse navbar-collapse\\" id=\\"navbarColor01\\"><ul class=\\"navbar-nav mr-auto\\"><li class=\\"nav-item active\\"><a class=\\"nav-link\\" href=\\"/dashboard\\">DASHBOARD<span class=\\"sr-only\\">(current)</span></a></li><li class=\\"nav-item active\\"><a class=\\"nav-link\\" href=\\"/browse\\">BROWSE<span class=\\"sr-only\\">(current)</span></a></li><li class=\\"nav-item active\\"><a class=\\"nav-link\\" href=\\"/upload\\">UPLOAD<span class=\\"sr-only\\">(current)</span></a></li><li class=\\"nav-item active\\"><a class=\\"nav-link\\" href=\\"/dashboard\\">SIGN OUT<span class=\\"sr-only\\">(current)</span></a></li><li class=\\"nav-item active\\"><a class=\\"nav-link\\" href=\\"/\\">ABOUT <span class=\\"sr-only\\">(current)</span></a></li></ul></div></div>"`;
